app/src/main/java/com/itsaky/androidide/actions/filetree/RenameAction.kt

```diff
@@ -65,29 +68,37 @@ class RenameAction(context: Context, override val order: Int) :
       dialogInterface,
       _ ->
       dialogInterface.dismiss()
-      val name: String = binding.name.editText!!.text.toString().trim()
-      val renamed = name.length in 1..40 && FileUtils.rename(file, name)
-      flashMessage(
-        if (renamed) com.itsaky.androidide.resources.R.string.renamed
-        else com.itsaky.androidide.resources.R.string.rename_failed,
-        if (renamed) FlashType.SUCCESS else FlashType.ERROR
-      )
-      if (!renamed) {
-        return@setPositiveButton
-      }
+      actionScope.launchAsyncWithProgress(
+          configureFlashbar = { builder, cancelChecker ->
+            builder.message(com.itsaky.androidide.resources.R.string.please_wait)
+          },
+          action = { _, _ ->
+            val name: String = binding.name.editText!!.text.toString().trim()
+            val renamed = name.length in 1..40 && FileUtils.rename(file, name)
 
-      notifyFileRenamed(file, name, context)
+            withContext(Dispatchers.Main) {
+              flashMessage(
+                  if (renamed) com.itsaky.androidide.resources.R.string.renamed
+                  else com.itsaky.androidide.resources.R.string.rename_failed,
+                  if (renamed) FlashType.SUCCESS else FlashType.ERROR)
+              if (!renamed) {
+                return@withContext
+              }
 
-      if (lastHeld != null) {
-        val parent = lastHeld.parent
-        parent.deleteChild(lastHeld)
-        val node = TreeNode(File(file.parentFile, name))
-        node.viewHolder = FileTreeViewHolder(context)
-        parent.addChild(node)
-        requestExpandNode(parent)
-      } else {
-        requestFileListing()
-      }
+              notifyFileRenamed(file, name, context)
```

Comment:
You can safely move `notifyFileRenamed` outside of the `Main` context. For `notifyFileRenamed`, EventBus will dispatch the event on the event receiver's preferred thread.